### PR TITLE
Add documentation for GEMM example

### PR DESCRIPTION
**Pull request by harsh-nod**
This PR adds documentation describing
how to implement GEMMs in tkw and in the
process explains the different primitives
and concepts in tkw.

### CODE REVIEW

File: shark_turbine/kernel/wave/docs/gemm_example.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
+# Matrix Multiplication
+
+Let's assume we want to compute the following matrix multiplication.
+$$ C = A \times B^{T} $$
```
Comment (Hardcode84) [inline shark_turbine/kernel/wave/docs/gemm_example.md:4]: Formatting is broken here: https://github.com/harsh-nod/iree-turbine/blob/dox/shark_turbine/kernel/wave/docs/gemm_example.md

File: shark_turbine/kernel/wave/docs/gemm_example.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+```python
+constraints: list[tkw.Constraint] = [tkw.WorkgroupConstraint(M, BLOCK_M, 0)]
+```
+
+This reads as "distribute dimension `M` with a tile size of `BLOCK_M` along workgroup dimension 0". So if we have a tensor of shape `[M, N]`, then with this constraint, each workgroup would be operating on a tile of size `[BLOCK_M, N]`.
+
+Once we have a workgroup constraint in place, we add further wave constraints on the same dimension.
+
+```
```
Comment (Hardcode84) [inline shark_turbine/kernel/wave/docs/gemm_example.md:128]: nit: missing ```python

File: shark_turbine/kernel/wave/docs/gemm_example.md (hunk 1 above)
Comment (harsh-nod) [reply shark_turbine/kernel/wave/docs/gemm_example.md:4]: thanks, will fix.

File: shark_turbine/kernel/wave/docs/gemm_example.md (hunk 2 above)
Comment (harsh-nod) [reply shark_turbine/kernel/wave/docs/gemm_example.md:128]: thanks, will add.

**Review by Hardcode84 — APPROVED** [review]
Overall looking good. We probably need a separate section on our `TestLaunchContext` harness and symbol binding.

**Pull request merged**